### DISCUSSION TITLE
Folder structure of ToDometerKotlinMultiplatform vs Folder structure of compose multiplatform template project

### DISCUSSION THREAD

**Discussion by shriharsha-srb** [serbelga/Todometer-KMP#358] [General]
I have a question regarding folder structure of this project.

If you follow the official [template](https://github.com/JetBrains/compose-multiplatform-template) for a compose multiplatform project, it has an `iosApp` folder, which is an Xcode project that builds into an iOS application. It depends on and uses the shared module as a CocoaPods dependency. 

I want to understand following if its okay :
1. Is it really needed to follow template folder structure? I assume answer is No as **ToDometerKotlinMultiplatform** already works without it and i see configurations done in `build.gradle.kts` of **iosComposeUi** module.
2. What are the advantages of template folder structure vs this project folder structure.
3. In what cases I might have to add `iosApp` module to this project in future?

**Comment by serbelga** [comment]
Hi, initially I created this project following that structure but then I used a kotlin gradle module instead of an XCode project for the ios app following some example I saw in the [compose-multiplatform](https://github.com/JetBrains/compose-multiplatform/tree/master) repository.
My idea is to explore this again to see if it is compatible with the current module structure in common, and check the advantages and disadvantages of it.

**Comment by shriharsha-srb** [comment]
Also in this current folder structure, if i want to add iOS specific code such as local notifications, is it possible ? Also if i want to introduce FCM, it should be handled by platform specific code. Not sure how to handle this in this project structure. Please let me know if you have any idea. 

I want to understand how to enable Firebase push notifications and its deeplink handling in each of the platforms.

**Reply by shriharsha-srb** [reply]
Hello thanks for this awesome repo. Just wanted to check once again regarding above question. Is there a way to include ios xcode project similar to template ? I need it as this will help me to create platform specific features like video player and push notifications.

**Reply by serbelga** [reply]
Hi, I have included the Xcode project in app-ios. You can see the changes here:
https://github.com/serbelga/Todometer-KMP/pull/414
Now, in main branch you can see the app-ios folder which contains a main.kt file and the Xcode project
https://github.com/serbelga/Todometer-KMP/tree/main/app-ios
